src/openrct2-ui/interface/Graph.cpp

```diff
@@ -182,10 +183,131 @@ static void graph_draw_line_b_money32(
     }
 }
 
+struct financial_tooltip_info
+{
+    const CoordsXY coords;
+    const money32 money{};
+};
+
+struct Rect
+{
+    const ScreenCoordsXY LeftTop;
+    const ScreenCoordsXY RightBottom;
+
+    int32_t GetLeft() const
+    {
+        return LeftTop.x;
+    }
+    int32_t GetTop() const
+    {
+        return LeftTop.y;
+    }
+    int32_t GetRight() const
+    {
+        return RightBottom.x;
+    }
+    int32_t GetBottom() const
+    {
+        return RightBottom.y;
+    }
+    int32_t GetWidth() const
+    {
+        return RightBottom.x - LeftTop.x;
+    }
+    int32_t GetHeight() const
+    {
+        return RightBottom.y - LeftTop.y;
+    }
+    bool Contains(const CoordsXY& coords) const
+    {
+        return coords.x >= GetLeft() && coords.x <= GetRight() && coords.y >= GetTop() && coords.y <= GetBottom();
+    }
+    bool Contains(const ScreenCoordsXY& coords) const
+    {
+        return Contains(CoordsXY(coords.x, coords.y));
+    }
+};
+
+static constexpr auto chartMaxDataCount = 64;
+static constexpr auto chartDataWidth = 6;
+static constexpr auto chartMaxIndex = chartMaxDataCount - 1;
+static constexpr auto chartMaxWidth = chartMaxIndex * chartDataWidth;
+static constexpr auto chartMaxHeight = 164;
+static constexpr auto cursorXOffset = 3;
+static constexpr auto defaultDashedLength = 2;
+
+static int32_t indexForCursorAndHistory(const int32_t historyCount, const int32_t cursorX, const int32_t chartX)
+{
+    const auto offsettedCursorX = cursorX + cursorXOffset;
+    return (historyCount - 1) - (offsettedCursorX - chartX) / chartDataWidth;
+}
+
+static const CoordsXY coordsForHistoryIndex(
```

Comment:
A lot of files still use the old lower case with underscore function names but our newer code uses TitleCase. This follows neither convention so I suggest making it Title Case.

Reply:
Should I make all the new functions TitleCase, or just the ones that don't follow either convention?

Reply:
It's up to you. I don't mind.